components/script/dom/eventtarget.rs

```diff
@@ -400,12 +407,25 @@ impl EventTarget {
                     entries.push(EventListenerEntry {
                         phase: ListenerPhase::Bubbling,
                         listener: EventListenerType::Inline(listener),
+                        once: false
                     });
                 }
             },
         }
     }
 
+    pub fn remove_listener_if_once(&self, ty: &Atom, listener: Rc<EventListener>) {
+        let mut handlers = self.handlers.borrow_mut();
+        let entry = handlers.get_mut(ty);
+
+        let listener = EventListenerType::Additive(listener.clone());
+        for entry in entry {
```

Comment:
I'm wondering if the naming would be confused? 🤔 How about naming the value from `handlers.get_mut` as `entries` and for-loop with `entry`?

```diff
- let entry = handlers.get_mut(ty);
+ let entries = handlers.get_mut(ty);
```

```diff
- for entry in entry
+ for entry in entries {
```

Reply:
Can't you just do `for entry in handlers.get_mut(ty)`?